mllib/src/test/scala/org/apache/spark/ml/feature/RModelFormulaSuite.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.feature
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.ml.param.ParamsSuite
+import org.apache.spark.mllib.linalg.Vectors
+import org.apache.spark.mllib.util.MLlibTestSparkContext
+import org.apache.spark.mllib.util.TestingUtils._
+
+class RFormulaModelSuite extends SparkFunSuite with MLlibTestSparkContext {
+  test("params") {
+    ParamsSuite.checkParams(new RModelFormula())
+  }
+
+  test("parse simple formulas") {
+    def check(formula: String, response: String, terms: Seq[String]) {
+      new RModelFormula().setFormula(formula)
```

Comment:
Whether to test private class or not might result much longer discussion:) In MLlib, usually we expose few public APIs, while the implementation might consist of several pieces. It is useful to test each piece individually though they are not public. For example, in ALS, https://github.com/apache/spark/blob/master/mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala#L53, it is hard to make useful unit test without unit testing individual components.